### PR TITLE
chore: download android driver from the cdn

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/playwright-core/src/server/android/android.ts [hunk 1]
```diff
@@ -186,8 +187,13 @@ export class AndroidDevice extends SdkObject {
       await this.shell(`cmd package uninstall com.microsoft.playwright.androiddriver.test`);
 
       debug('pw:android')('Installing the new driver');
-      for (const file of ['android-driver.apk', 'android-driver-target.apk'])
-        await this.installApk(await fs.promises.readFile(require.resolve(`../../../bin/${file}`)));
+      const executable = registry.findExecutable('android')!;
+      for (const file of ['android-driver.apk', 'android-driver-target.apk']) {
+        const fullName = path.join(executable.directory!, file);
+        if (!fs.existsSync(fullName))
+          throw new Error('Please install Android driver apk using `npx playwright install android`');
```
Comment (dgozman) [inline packages/playwright-core/src/server/android/android.ts:194]:
```suggestion
          throw new Error('Please install Android driver apk using `${buildPlaywrightCLICommand(sdkLanguage, 'install android')}`');
```

File: packages/playwright-core/src/server/registry/index.ts [hunk 2]
```diff
@@ -743,7 +761,8 @@ export class Registry {
   }
 
   private _downloadURLs(descriptor: BrowsersJSONDescriptor): string[] {
-    const downloadPathTemplate: string|undefined = (DOWNLOAD_PATHS as any)[descriptor.name][hostPlatform];
+    const paths = (DOWNLOAD_PATHS as any)[descriptor.name];
+    const downloadPathTemplate: string|undefined = paths[hostPlatform] || paths['<unknown>'];
```
Comment (dgozman) [inline packages/playwright-core/src/server/registry/index.ts:765]: I think this has a side-effect when downloading browsers, probably for unknown linux versions?

File: tests/android/playwright.config.ts [hunk 3]
```diff
@@ -67,6 +67,7 @@ config.projects.push({
     browserName: 'chromium',
   },
   testDir: path.join(testDir, 'page'),
+  testMatch: /page-check/,
```
Comment (mxschmitt) [inline tests/android/playwright.config.ts]: revert that

File: packages/playwright-core/src/server/registry/index.ts (hunk 2 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/registry/index.ts:765]: I don't think so?

File: packages/playwright-core/src/server/android/android.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/android/android.ts:194]: I don't think your suggestion compiles.

**Review by mxschmitt — APPROVED** [review]
lgtm but unknown -> default/fallback

**Pull request merged**